### PR TITLE
Smart with LSI raid controller

### PR DESCRIPTION
**Pull request by nedjitef**
I experienced some problems with the megacli "submodule":
1. when there is more than one controller, only the data of the first one can be gathered
2. when there is some SCSI compatible device directly attached, `/dev/sg0` may be something else than the raid controller (some hard drive or DVD drive connected to a SAS/SATA compatible port)

So we just added support for storcli and determine the sg device dynamically.

I know, this may be considered not to be a bugfix, because it adds a feature, but storcli support could be a benefit after all.

### CODE REVIEW

File: agents/plugins/smart [hunk 1]
```diff
@@ -62,6 +62,43 @@ megaraid_info()
...
 
+storcli_info()
+{
+    $1 /call show all | while read -r line; do
+        case "$line" in
+            "Controller = "*)
+                CTRL=$( echo "$line" | cut -d ' ' -f 3 )
+            ;;
+            "PCI Address = "*)
+                PCIA=$( echo "$line" | cut -d ' ' -f 4 | (IFS=: read -r _d _c _b _a; echo ${_a}${_b}:$_c:$_d) )
```
Comment (nedjitef) [inline agents/plugins/smart:73]: I'm not sure about this method to determine the PCI bus address.
It works for us so far, but we aren't sure, it's guaranteed to work yet.

**Comment by jherbel** [comment]
Hi, thank you for this PR. As you said, storcli could be quite beneficial, however, unfortunately we cannot accept this PR at this point in time, since this more than a bugfix.

**Comment by nedjitef** [comment]
So we would have to distribute our own version of the plugin for the hosts, we monitor?
I'm not sure, adding a second `<<<smart>>>` section would be a benefit.

Means, we would need to copy the smart checks, plugins etc and distribute a plugin?

**Comment by nedjitef** [comment]
I mean, we don't install megacli on new installation and even if we woud,
we are already left with hundreds of unmonitored HDDs and SSDs.

Is there an alternative on the way?

**Comment by jherbel** [comment]
You can simply replace the original plugin with your own version in the local site hierarchy. Put the modified version of the plugin under ~/local/share/check_mk/agents/plugins/. Then, Checkmk will use your version of the plugin and you can distribute it via the backery etc. If you wish, you can also package this as an MKP, that is up to you.

**Comment by nedjitef** [comment]
So you suggest to overwrite the existing agent plugin completely?

I thought about that, but I didn't want to submit a plugin to exchange.checkmk.com, that overrides builtin plugins.
This felt like the only way to contribute, that made sense to me. This just can't be an edge case.

**Comment by jherbel** [comment]
There is nothing wrong with overwriting the existing agent plugin. I don't know if this is an edge case or not, but independently of that, it is totally fine for an MKP on the exchange to overwrite a builtin plugin.

**Comment by nedjitef** [comment]
Just in case someone with the same problem finds this PR:

Works for me: https://exchange.checkmk.com/p/smart-patch

**Pull request closed without merging**